torch_xla/distributed/spmd/xla_sharding.py

```diff
@@ -16,24 +16,33 @@
 import numpy as np
 import functools
 import itertools
-from typing import Tuple, Union, List, Sequence, Any, Optional, Set
+from typing import Union, Sequence, Any, Optional
 from enum import IntEnum
 
 from torch.amp import custom_fwd, custom_bwd
 
+PartitionSpec = tuple[Union[tuple[Union[int, str], ...], int, str, None], ...]
+"""PartitionSpec describes the sharding of a tensor.
+
+Specifically, it is a tuple where each element references one or more device mesh
+axes, and instructs to shard the corresponding dimension of the input tensor over
```

Comment:
Just to make sure I understand what this is saying:  Is the first dimension of the input tensor sharded across the first element of the tuple, and so on? If so, we can say that. "... it is a tuple of one or more device mesh axes that describes how to shard the input tensor. For example, the first element of the tuple is sharded across the axis described in the first element of the tuple and so on."

Reply:
yes